eip_7594/src/lib.rs

```diff
@@ -343,6 +379,57 @@ fn kzg_commitment_inclusion_proof<P: Preset>(
     proof
 }
 
+/**
+ * Return the sample count if allowing failures.
+ *
+ * This helper demonstrates how to calculate the number of columns to query per slot when
+ * allowing given number of failures, assuming uniform random selection without replacement.
+*/
+pub fn get_extended_sample_count(allowed_failures: u64) -> u64 {
+    // check that `allowed_failures` within the accepted range [0 -> NUMBER_OF_COLUMNS // 2]
+    assert!((0..(NumberOfColumns::U64 / 2)).contains(&allowed_failures));
+
+    // missing chunks for more than a half is the worst case
+    let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
```

Comment:
Is the assertion correct here? Given that `worst_case_missing` is defined as `NumberOfColumns::U64 / 2 + 1`, the assertion should match this range:

```rust
let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
assert!((0..(worst_case_missing)).contains(&allowed_failures));
```

More importantly, using assert! for runtime validation can cause a panic, which isn't ideal for production code. The type checker doesn’t flag this as an issue, but a panic on failure isn’t appropriate for robust applications. Consider returning a `Result` (or `Option`) to handle errors gracefully:

```rust
pub fn get_extended_sample_count(allowed_failures: u64) -> Result<u64, &'static str> {
    let worst_case_missing = NumberOfColumns::U64 / 2 + 1;
    if !(0..(worst_case_missing)).contains(&allowed_failures) {
        return Err("allowed_failures out of range");
    }
    // Continue with the function logic...
}
```

Reply:
or 
```
    if allowed_failures >= worst_case_missing {
        return Err("allowed_failures out of range");
    }